### PR TITLE
Add frogger game task

### PR DESCRIPTION
**Pull request by AriadnaKravchuk**
### Classic Frogger Game

[Demo](https://ariadnakravchuk.github.io/frontend-nanodegree-arcade-game) |
[Code base](https://github.com/AriadnaKravchuk/frontend-nanodegree-arcade-game)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

**Comment by AriadnaKravchuk** [comment]
I completed self-checks

File: submissions/AriadnaKravchuk/frogger-game/app.js [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+
+function getNumFromArea(min, max) {
+    return Math.random() * (max - min) + min;
+}
+
+Array.prototype.getNumFromArray = function () {
+    return this[Math.floor(Math.random() * this.length)];
+};
+
+function changeChar(player) {
```
Comment (OleksiyRudenko) [inline submissions/AriadnaKravchuk/frogger-game/app.js:63]: Avoid abbreviations. We write variable/function names once, then IDE helps to autcomplete.
Does `Char` mean `character`, `charcoal`, `Charizard`? (a rhetoric question)

File: submissions/AriadnaKravchuk/frogger-game/app.js [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+    }
+
+    this.collision();
+};
+
+Enemy.prototype.render = function () {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+Enemy.prototype.collision = function () {
```
Comment (OleksiyRudenko) [inline submissions/AriadnaKravchuk/frogger-game/app.js:111]: By convention, function names should start with a verb.

File: submissions/AriadnaKravchuk/frogger-game/app.js [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+    } else if (keyPress === "up" && this.y !== BORDER.MAX_Y) {
+        checkRock(this, allRocks);
+    } else if (keyPress === "right" && this.x !== BORDER.MAX_X) {
+        this.x += STEP.X;
+    } else if (keyPress === "down" && this.y !== BORDER.MIN_Y) {
+        this.y += STEP.Y;
+    }
+};
+
+function checkRock(player, allRocks) {
```
Comment (OleksiyRudenko) [inline submissions/AriadnaKravchuk/frogger-game/app.js:154]: Why this is not a class method?
Or why `collision` is a part of `Enemy` class?

**Review by OleksiyRudenko — CHANGES_REQUESTED** [review]
@AriadnaKravchuk very well done!
We could skip minor naming issues, but one thing with objects behaviour scoping is important.

**Comment by AriadnaKravchuk** [comment]
> @AriadnaKravchuk very well done! We could skip minor naming issues, but one thing with objects behaviour scoping is important.

I changed functions names and made collision checking like player methods.

**Review by OleksiyRudenko — APPROVED** [review]
@AriadnaKravchuk great job!

**Pull request merged**